### DISCUSSION TITLE
Thread safety of ILogger<T>

### DISCUSSION THREAD

**Discussion by EnricoMassone** [dotnet/runtime#50695] [Q&A] [answered]
According to [this documentation](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-5.0#framework-provided-services) the ILogger\<T\> interface is automatically registered as a singleton with the DI container when using the ASP.NET core project templates. 

Givent that, can I safely assume that any available (and properly implemented) implementation of ILogger\<T\> is actually thread safe ?

**Comment by davidfowl** [comment, marked as answer]
Yes it's thread safe.

**Reply by EnricoMassone** [reply]
@davidfowl thanks!

**Comment by ghost** [comment]
Tagging subscribers to this area: @eerhardt, @maryamariyan
See info in [area-owners.md](https://github.com/dotnet/runtime/blob/main/docs/area-owners.md) if you want to be subscribed.
<details>
<summary>Issue Details</summary>
<hr />

According to [this documentation](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-5.0#framework-provided-services) the ILogger\<T\> interface is automatically registered as a singleton with the DI container when using the ASP.NET core project templates. 

Givent that, can I safely assume that any available (and properly implemented) implementation of ILogger\<T\> is actually thread safe ? 

<table>
  <tr>
    <th align="left">Author:</th>
    <td>EnricoMassone</td>
  </tr>
  <tr>
    <th align="left">Assignees:</th>
    <td>-</td>
  </tr>
  <tr>
    <th align="left">Labels:</th>
    <td>

`area-Extensions-DependencyInjection`, `untriaged`

</td>
  </tr>
  <tr>
    <th align="left">Milestone:</th>
    <td>-</td>
  </tr>
</table>
</details>
